### PR TITLE
fix(select): Options within an opt group now sync to select value when slot changes

### PR DESCRIPTION
**Pull request by micahjones13**
## Brief Description

When dynamically adding/removing `rux-option`'s from a `rux-option-group`, the value of the `rux-select` would not properly sync to a `rux-option` within the `rux-option` group. 
The method to sync these values wasn't being called when a `rux-option-group` emitted it's changed event.

I left the `src/index` with an example of the fix in action for convenience  of review. Check out the astro-stencil preview to see it. 
Here's a sandbox version of the bug without the fix as well: https://codesandbox.io/s/misty-moon-0c2nfm?file=/index.html
## JIRA Link

https://rocketcom.atlassian.net/browse/ASTRO-6186

## Related Issue
https://github.com/RocketCommunicationsInc/astro/issues/1125
## General Notes

Added 2 tests for adding/removing `rux-option`'s from a `rux-option-group` and having the pre-selected value persist. 

## Motivation and Context

Allows for pre-selected values on `rux-select` to persist when changes to `rux-options` within a `rux-option-group` are made.

## Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [x] I have added tests to cover my changes.
- [ ] Regressions are passing and/or failures are documented
- [ ] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-select/test/select.spec.ts [hunk 1]
```diff
@@ -74,7 +74,88 @@ test.describe('Select', () => {
...
+                      let newOption = document.createElement('rux-option')
+                      newOption.label = hero.name
+                      newOption.value = hero.value
+                      optGroup.appendChild(newOption)
+                  })
+        })
+        `,
+        })
+        //selected value should be flash
+        const sel = page.locator('rux-select').locator('select')
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-select/test/select.spec.ts:110]: you don't have to chain locators here. you can just do `page.locator('rux-select select')` and playwright will reach in the shadow dom.

Comment (micahjones13) [reply packages/web-components/src/components/rux-select/test/select.spec.ts:110]: Good to know!

File: packages/web-components/src/components/rux-select/test/select.spec.ts [hunk 2]
```diff
@@ -74,7 +74,88 @@ test.describe('Select', () => {
...
+        await expect(sel).toHaveValue('flash')
+        const btn = page.locator('rux-button')
+        //click btn to add more options inside opt group
+        await btn.click()
+        await page.waitForChanges()
+        //selected value should still be flash
+        await expect(sel).toHaveValue('flash')
+    })
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-select/test/select.spec.ts:118]:
```suggestion
    test('Options that are dynamically added can be synced to select', async ({
        page,
    }) => {
        const template = `
            <rux-select value="flash">
                <rux-option value="" label="Select"></rux-option>
                <rux-option-group label="Group">
                    <rux-option value="flash" label="Flash"></rux-option>
                </rux-option-group>
            </rux-select> 
        `
        await page.setContent(template)

        const select = await page.locator('rux-select select')
        await expect(select).toHaveValue('flash')

        // Add Batman to Option Group
        const optionGroupEl = await page.locator('rux-option-group')
        await optionGroupEl.evaluate((el) => {
            const newOption = document.createElement('rux-option')
            newOption.label = 'batman'
            newOption.value = 'batman'
            el.appendChild(newOption)
        })

        await page.waitForChanges()
        await expect(select).toHaveValue('flash')
    })
```

slightly cleaner way without having to add a bunch of stuff in an initscript

File: packages/web-components/src/index.html [hunk 3]
```diff
@@ -19,5 +19,31 @@
...
+            btn.addEventListener('click', () => {
+                init.map((hero) => {
+                    let newOption = document.createElement('rux-option')
+                    newOption.label = hero.name
+                    newOption.value = hero.value
+                    optGroup.appendChild(newOption)
+                })
+            })
+        </script>
+    </body>
```
Comment (markacianfrani) [inline packages/web-components/src/index.html:48]:
```suggestion
    <body></body>
```

cleanme

**Review by markacianfrani — APPROVED** [review]
some nits but LGTM!

File: packages/web-components/src/components/rux-select/test/select.spec.ts (hunk 1 above)
Comment (micahjones13) [inline packages/web-components/src/components/rux-select/test/select.spec.ts]:
```suggestion
        const sel = page.locator('rux-select select')
```

File: packages/web-components/src/components/rux-select/test/select.spec.ts [hunk 4]
```diff
@@ -74,7 +74,74 @@ test.describe('Select', () => {
...
+            const optGroup = document.querySelector('rux-option-group')
+            const btn = document.querySelector('rux-button')
+            btn.addEventListener('click', () => {
+              let options = optGroup.querySelectorAll('rux-option')
+              options[options.length - 1].remove()
+            })
+      `,
+        })
+        //selected value should be flash
+        const sel = page.locator('rux-select').locator('select')
```
Comment (micahjones13) [inline packages/web-components/src/components/rux-select/test/select.spec.ts:132]:
```suggestion
        const sel = page.locator('rux-select select')
```

**Pull request merged**